timemachine/md/minimizer.py

```diff
@@ -374,25 +385,30 @@ def val_and_grad_fn_bfgs(x_local_flattened):
     x_local_final_flat = res.x
     x_local_final = x_local_final_flat.reshape(x_local_shape)
 
+    U_final, grad_final = val_and_grad_fn_bfgs(x_local_final_flat)
+    forces = -grad_final.reshape(x_local_shape)
+    per_atom_force_norms = np.linalg.norm(forces, axis=1)
+
     if verbose:
-        U_final, grad_final = val_and_grad_fn_bfgs(x_local_final_flat)
         print(f"U(x_final) = {U_final:.3f}")
-
         # diagnose worst atom
-        forces = -grad_final.reshape(x_local_shape)
-        per_atom_force_norms = np.linalg.norm(forces, axis=1)
         argmax_local = np.argmax(per_atom_force_norms)
         worst_atom_idx = local_idxs[argmax_local]
         print(f"atom with highest force norm after minimization: {worst_atom_idx}")
         print(f"force(x_final)[{worst_atom_idx}] = {forces[argmax_local]}")
-
         print("-" * 70)
 
+    # note that this over the local atoms only, as this function is not concerned
+    assert np.amax(per_atom_force_norms) < MAX_FORCE_NORM
```

Comment:
Nice -- this seems like a confident assertion, and better aligns with the intent of `local_minimize`.

(Don't necessarily want an assertion error if the user happens to pass in a minimized structure / if line search can't find a step that satisfies "sufficient decrease" conditions, but do always want an assertion error if the result is unsuitable to use as an initial condition for MD.)